packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js

```diff
@@ -108,28 +109,60 @@ export default class TrafficController {
       default:
         after.setDate(after.getDate() - 1);
     }
+    this.getAllTraffic(null, after.toISOString(), before, this.subnet);
+  }
+
+  getAllTraffic(cursor, after, before, subnet) {
     const params = {
-      after: after.toISOString(),
+      after,
       before,
-      subnet: this.subnet,
+      subnet,
     };
     this.networkSecurityService
-      .getTraffic(params)
-      .then(({ data }) => {
+      .getAllTraffic({
+        cursor,
+        params,
+        pageSize: this.PAGE_SIZE,
+      })
+      .then((response) => {
+        const { data } = response;
         if (data.message) {
           this.Alerter.error(
             this.$translate.instant('network_security_dashboard_events_error'),
             'network_security_error',
           );
           return data;
         }
-        this.results = data;
-        this.resume(after.toISOString(), before, this.subnet);
-        this.loadGraph();
-        return data;
-      })
-      .finally(() => {
-        this.isLoading = false;
+        if (data) {
+          if (!this.results) {
+            this.results = data;
+          } else {
+            this.results.timestamps = this.results.timestamps.concat(
+              data.timestamps,
+            );
+            this.results.bps.dropped = this.results.bps.dropped.concat(
+              data.bps.dropped,
+            );
+            this.results.bps.passed = this.results.bps.passed.concat(
+              data.bps.passed,
+            );
+            this.results.pps.dropped = this.results.pps.dropped.concat(
+              data.pps.dropped,
+            );
+            this.results.pps.passed = this.results.pps.passed.concat(
+              data.pps.passed,
+            );
+          }
+        }
+        if (response.cursor.next) {
+          this.getAllTraffic(response.cursor.next, after, before, subnet);
+        } else {
```

Comment:
API cursor is still KO ? You can't use pagination ?

Reply:
I use this API to retrieve data to display them into a graph component. So I need to retrieve all data to build the graph.